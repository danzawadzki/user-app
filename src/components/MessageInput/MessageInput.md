Default MessageInput with all props:

```jsx
<MessageInput
	value="Test value"
	placeholder="Add a comment..."
	onChange={e => console.log(e.target.value)}
	onSubmit={e => {
		e.preventDefault();
		console.log(e);
	}}
/>
```
