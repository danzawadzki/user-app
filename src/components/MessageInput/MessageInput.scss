@import '../../assets/styles/colors';
@import '../../assets/styles/minor';

.MessageInput {
	margin-top: 2rem;
	&__input {
		border: none;
		color: $blue--light;
		width: 100%;
		padding: 1rem 0;
		font-size: 1.8rem;
		border-bottom: 2px solid $app__borders;
		transition: $transition--default;

		&::placeholder {
			color: $blue--light;
		}

		&:focus {
			border-bottom: 2px solid $brand--accent;
		}
	}
}
