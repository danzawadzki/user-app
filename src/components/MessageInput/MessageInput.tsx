import * as React from 'react';
import './MessageInput.scss';

export interface IMessageInput {
	/** Component id for onChange method */
	id: string;
	/** Controlled form input value */
	value?: string;
	/** Input placeholder */
	placeholder?: string;
	/** Handler to onChange event */
	onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
	/** Handler to onSubmit event */
	onSubmit: (e: React.FormEvent<HTMLFormElement>) => void;
}

/**
 * MessageInput component.
 *
 * @version 1.0.0
 */
const MessageInput = ({
	value = '',
	id,
	placeholder = 'Type...',
	onChange,
	onSubmit
}: IMessageInput) => {
	return (
		<form className="MessageInput" onSubmit={onSubmit}>
			<input
				id={id}
				className="MessageInput__input"
				value={value}
				placeholder={placeholder}
				onChange={onChange}
			/>
		</form>
	);
};

export default MessageInput;
