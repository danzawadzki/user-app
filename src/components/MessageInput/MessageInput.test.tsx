import React from 'react';
import { shallow } from 'enzyme';
import MessageInput from './MessageInput';

describe('MessageInput component: ', () => {
	//Shallow rendered component
	const value = 'Test';
	const id = 'TestInput';
	const placeholder = 'Type comment...';
	const onChange = (e: React.ChangeEvent<HTMLInputElement>) => console.log(e);
	const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
		e.preventDefault();
		console.log(e);
	};

	const component = shallow(
		<MessageInput
			id={id}
			value={value}
			placeholder={placeholder}
			onChange={onChange}
			onSubmit={onSubmit}
		/>
	);

	it('should render match snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('should contain value and placeholder', () => {
		expect(component.html()).toContain(value);
		expect(component.html()).toContain(placeholder);
	});
});
