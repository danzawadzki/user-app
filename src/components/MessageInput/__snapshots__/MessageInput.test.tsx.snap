// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput component:  should render match snapshot 1`] = `
<form
  className="MessageInput"
  onSubmit={[Function]}
>
  <input
    className="MessageInput__input"
    id="TestInput"
    onChange={[Function]}
    placeholder="Type comment..."
    value="Test"
  />
</form>
`;
